.github/pull_request_template.md

```diff
@@ -30,48 +32,30 @@ of the person or team responsible for reviewing proposed changes. -->
-- [ ] Class methods are `lowerCamelCase`
-- [ ] Method arguments are `lowerCamelCase`
-- [ ] Normal variables are `lowerCamelCase`
-- [ ] Constants are `k_UpperCamelCase`
-- [ ] Global statics are `s_UpperCamelCase`
-- [ ] Free Functions are `UpperCamelCase`
-- [ ] Macros are `ALL_UPPER_SNAKE_CASE`
-- [ ] Unit test will test data output from the filter.
-- [ ] Reused strings should be constants in an anonymous namespace
-- [ ] If parallelization is used, proper use of the abstracted complex classes are used. Using TBB specifically in a filter should be frowned upon unless for a really good reason.
+Naming of variables should descriptive where needed. Loop Control Variables can use `i` if warranted. Most of these conventions are enforced through the clang-tidy and clang-format configuration files. See the file `complex/docs/Code_Style_Guide.md` for a more in depth explanation.
+
 
 ## Filter Checklist
 
-- [ ] Parameters should be generally broken down into "Input Parameters", "Required Data Objects", "Created Data Objects". There can be exceptions to this.
-- [ ] ChoicesParameter selections should be an enumeration defined in the filer header
-- [ ] Documentation copied from SIMPL Repo and updated (if necessary)
-- [ ] Parameter argument variables are k_CamelCase_Key
-- [ ] Parameter argument strings are lower_snake_case
-```
-static inline constexpr StringLiteral k_AlignmentType_Key = "alignment_type";
-```
+The help file `complex/docs/Porting_Filters.md` has documentation to help you port or write new filters. At the top is a nice checklist of items that should be noted when porting a filter.
+
 
 ## Unit Testing
-- [ ] 1 Unit test to instantiate the filter with the default arguments.
+
+The idea of unit testing is to test the filter for proper execution and error handling. How many variations on a unit test each filter needs is entirely dependent on what the filter is doing. Generally, the variations can fall into a few categories:
+
 - [ ] 1 Unit test to test output from the filter against known exemplar set of data
 - [ ] 1 Unit test to test invalid input code paths that are specific to a filter. Don't test that a DataPath does not exist since that test is already performed as part of the SelectDataArrayAction.
 
 ## Code Cleanup
 - [ ] No commented out code (rare exceptions to this is allowed..)
-- [ ] Filters should have both the Filter class and Algorithm class for anything beyond trivial needs
 - [ ] No API changes were made (or the changes have been approved)
 - [ ] No major design changes were made (or the changes have been approved)
 - [ ] Added test (or behavior not changed)
 - [ ] Updated API documentation (or API not changed)
 - [ ] Added license to new files (if any)
-- [ ] Added Python wrapping to new files (if any) as necessary
 - [ ] Added example pipelines that use the filter
-- [ ] Classes and methods are properly documented.
+- [ ] Classes and methods are properly documented
 
```

Comment:
```suggestion
- [ ] Make use of Execute functions in FilterUtitilites and ParallelAlgorithmUtilities wherever possible to remove boilerplate switches 
```
Thoughts on adding this?

Reply:
I think things like this should go in either the Style Guide or the Porting Guide. I think of the porting guide as the Tutorial type of doc where someone can go look for ideas on how to get something done or what design pattern to use to accomplish something.

I think the PR Template should point developers/reviewers to those documents.